File: tests/fixtures/typescript/clean_catch.ts
function risky(): string {
  return "ok";
}
function fallback(): string {
  return "fallback";
}
function optional(): void {}
function reportError(_e: unknown): void {}

function rethrowCatch() {
  try {
    risky();
  } catch (e) {
    throw e;
  }
}

function returnCatch() {
  try {
    return risky();
  } catch (e) {
    return null;
  }
}

function recoveryAssignCatch() {
  let result;
  try {
    result = risky();
  } catch (e) {
    result = fallback();
  }
  return result;
}

function intentCatch() {
  try {
    optional();
  } catch (e) {
    // Intentional: operation is optional, ignore failure
  }
}

function nonConsoleCallCatch() {
  try {
    risky();
  } catch (e) {
    reportError(e);
  }
}

File: tests/fixtures/rust/clean_catch.rs
use std::fmt;

#[derive(Debug)]
struct MyError;

impl fmt::Display for MyError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "boom")
    }
}

fn risky() -> Result<i32, MyError> {
    Err(MyError)
}

fn log_and_return() -> i32 {
    match risky() {
        Err(e) => {
            eprintln!("failed: {}", e);
            -1
        }
        Ok(v) => v,
    }
}

fn intent_arm() {
    match risky() {
        Err(e) => {
            // intentional: best-effort operation, ignore failure
        }
        Ok(v) => println!("{}", v),
    }
}

fn other_pattern_empty() {
    let x: Option<i32> = None;
    match x {
        None => {}
        Some(v) => println!("{}", v),
    }
}
